tests/ui/structs/default-field-values/non-exhaustive-ctor.stderr

```diff
@@ -0,0 +1,73 @@
+error[E0603]: unit struct `Priv` is private
+  --> $DIR/non-exhaustive-ctor.rs:20:39
+   |
+LL |     let _ = S { field: (), field1: m::Priv };
+   |                                       ^^^^ private unit struct
+   |
+note: the unit struct `Priv` is defined here
+  --> $DIR/non-exhaustive-ctor.rs:10:5
+   |
+LL |     struct Priv;
+   |     ^^^^^^^^^^^^
+
+error[E0603]: unit struct `Priv` is private
+  --> $DIR/non-exhaustive-ctor.rs:23:39
+   |
+LL |     let _ = S { field: (), field1: m::Priv, field2: m::Priv };
+   |                                       ^^^^ private unit struct
+   |
+note: the unit struct `Priv` is defined here
+  --> $DIR/non-exhaustive-ctor.rs:10:5
+   |
+LL |     struct Priv;
+   |     ^^^^^^^^^^^^
```

Comment:
This case is not relevant to *this* PR, but I'm looking on `resolve` to make it also suggest `..` if `Priv` is private but `field1` is public.

This will allow API designs where the construction of a public struct will always need `..`, protecting against future API breakage (in the same way that `non_exhaustive` is used in patterns).